### ISSUE TITLE
Consider removing package-lock.json from repo

### ISSUE THREAD

**Issue by limonte** [kenwheeler/cash#220]
`package-lock.json` is a joke:

- https://stackoverflow.com/q/45022048/1331425
- https://github.com/npm/npm/issues/17979

### Current behavior

```
git clone git@github.com:kenwheeler/cash.git
npm install

git status

On branch remove-package-lock
Your branch is up to date with 'origin/remove-package-lock'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   package-lock.json

```

`package-lock.json` is changed, nonsense which only developers of `npm/cli` can understand :) Other people seem to be [confused](https://github.com/npm/npm/issues/18103#issuecomment-370401935):

![image](https://user-images.githubusercontent.com/6059356/46935793-edd0ef00-d064-11e8-877c-456a425b50f2.png)

### Expected behavior

Git working tree is clean, `package-lock.json` is the same.

**Comment by fabiospampinato** [comment]
Yeah modifying package-_lock_.json is kind of weird, let me think more about this though 👍

**Comment by fabiospampinato** [comment]
I'll remove `package-lock.json` from `cash` (and all my other libraries), for the following reasons, some of which don't actually matter for `cash` because it has no dependencies:

- It can't be published, so it's only potentially useful while developing
- Since it can't be published it only locks my dependencies, but not those of the users who will be installing it, potentially preventing me from hitting some bugs due to broken dependencies updates
- It gets updated on `npm install` and everytime a new dependency is added
- It generate a bit of noise in the commits

The comment you quoted said:

> And please don't tell me I am suppose to remove the carets from my package.json otherwise I might as well consider that to be our new lock file.

Which is kinda flawed, since fixing the dependencies in your `package.json` doesn't prevent the dependencies of your dependencies to get updated, a lock file gives you that.

I think overall the way `npm` handled lock files is kind of a mess, for some reason they actually support  2 different files, `package-lock.json` and `npm-shrinkwrap.json`, the latter of which can actually be published.

I don't think we actually need a lockfile, if I'll change my mind I'll give `yarn.lock` a shot.
